### PR TITLE
[VL] Minor follow-ups for #6942

### PR DESCRIPTION
**Pull request by zhztheplayer**
Following #6942,

1. Remove prefix `org/apache/gluten` for library pathes in Jar
2. ~~A test to assert on the runtime error message when Gluten build is used on unsupported platform.~~

### CODE REVIEW

**Comment by zhztheplayer** [comment]
@wForget

**Review by wForget — APPROVED** [review]
Thanks, LGTM

**Comment by wForget** [comment]
Related failures

```
VeloxListenerApiSuite:
- Unsupported arch *** FAILED ***
  Expected exception java.lang.RuntimeException to be thrown, but no exception was thrown (VeloxListenerApiSuite.scala:27)

- csv scan *** FAILED ***
  org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 730.0 failed 1 times, most recent failure: Lost task 0.0 in stage 730.0 (TID 540) (f98bd9d27030 executor driver): java.lang.IllegalStateException: error loading native libraries: java.io.FileNotFoundException: unknown-arch/libarrow_dataset_jni.so
	at org.apache.arrow.dataset.jni.JniLoader.load(JniLoader.java:95)
```

File: backends-velox/src/test/scala/org/apache/gluten/backendsapi/velox/VeloxListenerApiSuite.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+package org.apache.gluten.backendsapi.velox
+
+import org.apache.gluten.test.VeloxBackendTestBase
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class VeloxListenerApiSuite extends AnyFunSuite {
+  test("Unsupported arch") {
+    System.setProperty("os.arch", "unknown-arch")
```
Comment (wForget) [inline backends-velox/src/test/scala/org/apache/gluten/backendsapi/velox/VeloxListenerApiSuite.scala:25]: This property needs to be reset

Comment (zhztheplayer) [reply backends-velox/src/test/scala/org/apache/gluten/backendsapi/velox/VeloxListenerApiSuite.scala:26]: Actually there is no easy way to run this test correctly with other tests in same process as we only load the libraries once.

I'll remove the test. It's not that important to place it in code base. I already verified the error locally.

**Review by wForget — APPROVED** [review]

**Review by PHILO-HE — APPROVED** [review]
Thanks!

**Pull request merged**